skimage/feature/corner_cy.pyx

```diff
@@ -145,6 +145,19 @@ def _corner_fast(np_floats[:, ::1] image, signed char n, np_floats threshold):
                 lower_threshold = curr_pixel - threshold
                 upper_threshold = curr_pixel + threshold
 
+                # High speed test for n >= 12
+                if n >= 12:
+                    speed_sum_b = 0
+                    speed_sum_d = 0
+                    for k in range(0, 16, 4):
+                        ring_pixel = image[i + rp[k], j + cp[k]]
```

Comment:
Thank you for the prompt response @spdfghi. It's beginning to make sense. A follow-up question:

Considering the offset arrays `rp` and `cp` have 16 elements, how do tehese lines make sense. Shouldn't `rp[16]` be out of bounds and return a nonsensical result? This is also an issue with the original implementation...

Reply:
you know it. Just to clarify.  I usually work with c++,  i don't know how pyx become lib. I will check letter, but i think testing code is just like this, with larger image:
`from skimage.feature import corner_fast, corner_peaks
square = np.zeros(1200, 1200))
square.astype(int)`

Reply:
@lagru i am not native English speaker, so i might misunderstood the question, as  range for in python is used like [a, b),  range(0, 16, 4) is 0, 4, 8, 12